ras_backstage/resources/sign_in/sign_in.py

```diff
@@ -53,12 +53,17 @@ def post():
         username = message_json.get('username')
         password = message_json.get('password')
 
-        # Obviously horrible, stopgap until uaa is implemented
-        if username == current_app.config['USERNAME'] and password == current_app.config['PASSWORD']:
-            # We're assuming that uaa will return an Oauth2 token though it's almost certain that
-            # this will change once we know exactly what is being returned.
-            logger.info("Authentication successful", user=username)
-            return make_response(jsonify({"token": "1234abc"}), 201)
+        if current_app.config.get('USE_UAA'):
+            logger.info('Retrieving sign-in details')
+            oauth2_token = uaa_controller.sign_in(username, password)
+            logger.info('Successfully retrieved sign-in details')
+            return make_response(jsonify({"token": oauth2_token}), 201)
         else:
-            logger.info("Authentication failed", user=username)
-            return make_response(jsonify({"error": "Username and/or password incorrect"}), 401)
+            #  TODO remove this once UAA fully deployed in all environments
+            if username == current_app.config['USERNAME'] and password == current_app.config['PASSWORD']:
+                # We're assuming that uaa will return an Oauth2 token though it's almost certain that
```

Comment:
:art: "We're almost certain that ... " what? Seems to be cut off :)

Reply:
We aren't assuming anything here @necrophonic 😉 will remove